When Kytos controller reconnect to the switch, the flows should be preserved

Hi,

When the Kytos controller disconnects from the switch and then reconnects, all the flows are removed and recreated, causing a traffic disruption. When the switch is reconnected to the controller, the consistency check should work only in the inconsistencies and not remove all to create.

The following end-to-end test is failing:
```
_______ TestE2EFlowManager.test_030_restart_kytos_should_preserve_flows ________

self = <tests.test_e2e_21_flow_manager.TestE2EFlowManager instance at 0x7fbbe41b9d88>

   def test_030_restart_kytos_should_preserve_flows(self):
       """Test if, after kytos restart, the flows are preserved on the switch
          flow table."""
       self.net.restart_kytos_clean()
       time.sleep(5)

       payload = {
           "flows": [
               {
               "priority": 10,
               "match": {
                   "in_port": 1,
                   "dl_vlan": 999
               },
               "actions": [
                   {
                   "action_type": "output",
                   "port": 2
                   }
               ]
               }
           ]
       }

       api_url = KYTOS_API + '/flow_manager/v2/flows/00:00:00:00:00:00:00:01'
       response = requests.post(api_url, data=json.dumps(payload),
                                headers={'Content-type': 'application/json'})
       assert response.status_code == 200
       data = response.json()
       assert 'FlowMod Messages Sent' in data['response']

       # wait for the flow to be installed
       wait_time = 20
       time.sleep(wait_time)

       # restart controller keeping configuration
       t1 = time.time()
       self.net.start_controller()
       self.net.wait_switches_connect()
       delta = time.time() - t1

       # wait for the flow to be installed
       time.sleep(wait_time)
       wait_time += wait_time

       s1 = self.net.net.get('s1')
       flows_s1 = s1.dpctl('dump-flows')
       assert len(flows_s1.split('\r\n ')) == 2
       for flow in flows_s1.split('\r\n '):
           match = re.search("duration=([0-9.]+)", flow)
           duration = float(match.group(1))
         assert duration >= wait_time + delta
E           assert 21.996 >= (40 + 10.281726837158203)

```

This issue can also be related to steps taken from Kytos when the switch connects. Many openflow controllers used to delete all flows when the switch has a ConnectionUP (not sure if this is the case for kytos)

--- a/main.py
+++ b/main.py
@@ -57,6 +57,9 @@ class Main(KytosNApp):
     @listen_to('kytos/of_core.handshake.completed')
     def resend_stored_flows(self, event):
         """Resend stored Flows."""
+        # if consistency check is enabled, it should take care of this
+        if CONSISTENCY_INTERVAL >= 0:
+            return
         switch = event.content['switch']
         dpid = str(switch.dpid)
         # This can be a problem because this code is running a thread

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -232,6 +232,7 @@ class TestMain(TestCase):
         self.napp._load_flows()
         mock_storehouse.assert_called()
 
+    @patch("napps.kytos.flow_manager.main.CONSISTENCY_INTERVAL", -1)
     @patch("napps.kytos.flow_manager.main.Main._install_flows")
     def test_resend_stored_flows(self, mock_install_flows):
         """Test resend stored flows."""

